File: engine/view_mode.py
class ViewMode:
    """
    Controla Scene View / Game View.
    """

    SCENE_VIEW = "SCENE_VIEW"
    GAME_VIEW = "GAME_VIEW"

    def __init__(self, game):
        self.game = game
        self.mode = self.SCENE_VIEW

    def toggle(self):
        if self.mode == self.SCENE_VIEW:
            self.mode = self.GAME_VIEW
            self.game.console.log("Vista: GAME VIEW", "ENGINE")
        else:
            self.mode = self.SCENE_VIEW
            self.game.console.log("Vista: SCENE VIEW", "ENGINE")

    def is_game_view(self):
        return self.mode == self.GAME_VIEW

    def is_scene_view(self):
        return self.mode == self.SCENE_VIEW

File: src/engine/view_mode.rs
#[derive(Debug, Clone, PartialEq, Eq)]
pub enum ViewModeKind {
    SceneView,
    GameView,
}

#[derive(Debug, Clone)]
pub struct ViewMode {
    pub mode: ViewModeKind,
}

impl Default for ViewMode {
    fn default() -> Self {
        Self {
            mode: ViewModeKind::SceneView,
        }
    }
}

impl ViewMode {
    pub const SCENE_VIEW: ViewModeKind = ViewModeKind::SceneView;
    pub const GAME_VIEW: ViewModeKind = ViewModeKind::GameView;

    pub fn toggle(&mut self) {
        self.mode = match self.mode {
            ViewModeKind::SceneView => ViewModeKind::GameView,
            ViewModeKind::GameView => ViewModeKind::SceneView,
        };
    }
}
